Sources/StreamChat/ChatClient.swift

```diff
@@ -404,6 +405,48 @@ public class ChatClient {
             completion()
         }
     }
+    
+    // MARK: - Listening for Client Events
+    
+    /// Subscribes to web-socket events of the specified event type.
+    ///
+    /// - Note: The handler is always called on the main thread.
+    ///
+    /// An example of observing connection status changes:
+    /// ```swift
+    /// client.subscribe(to: ConnectionStatusUpdated.self) { connectionEvent in
+    ///     switch connectionEvent.connectionStatus {
+    ///         case .connected:
+    ///           …
+    ///     }
+    /// }
```

Comment:
Nice 👍 Initially, I thought about creating a separate function per event, to avoid all the casting we do right now, but this is actually even better 👍